scan/enum.go

```diff
@@ -0,0 +1,100 @@
+package scan
+
+import (
+	"go/ast"
+	"strconv"
+	"unicode"
+	"strings"
+)
+
+func enumName(comments *ast.CommentGroup, typeSpec *ast.TypeSpec) (name string, isEnum bool) {
+	if comments != nil {
+		for _, cmt := range comments.List {
+			for _, ln := range strings.Split(cmt.Text, "\n") {
+				matches := rxEnumOverride.FindStringSubmatch(ln)
+				if len(matches) > 1 {
+					name = matches[1]
+					if name == "" {
+						return typeSpec.Name.String(), true
+					}
+					return strings.TrimSpace(name), true
+				}
+			}
+		}
+	}
+	return "", false
+}
+
+func upperSnakeCase(s string) string {
```

Comment:
Yes. We can ignore the complex objects, it is not widely used.
And for stings and integer, we could follow the type which is annotated.

```
// swagger:enum
type Level int

const (
	LEVEL_1 Level = 1
	LEVEL_2 Level = 2
	LEVEL_3 Level = 3
)
```
will be
```
type: "integer"
enum:
 - 1
 - 2
 - 3
```